packages/hub/plugin-loader.js

```diff
@@ -114,22 +115,23 @@ class PluginLoader {
 
   async _crawlPlugins(dir, outputPlugins, seen, includeDevDependencies, breadcrumbs) {
     log.trace("plugin crawl dir=%s, includeDevDependencies=%s, breadcrumbs=%j", dir, includeDevDependencies, breadcrumbs);
-    if (seen[dir]) {
-      if (seen[dir].attributes && seen[dir].attributes.includedFrom) {
+    let realdir = await realpath(dir);
+    let packageJSON = path.join(realdir, 'package.json');
+    let json = require(packageJSON);
```

Comment:
I think this eliminated an optimization -- we're going to re-parse the package.json for directories we've already seen, where before we hit the early `return`and skipped that.

Reply:
thanks, i'll fix that